src/semidiscretization/semidiscretization_coupled.jl

```diff
@@ -137,23 +140,36 @@ end
     @view u_ode[semi.u_indices[index]]
 end
 
+# Same as `foreach(enumerate(something))`, but without allocations.
+#
+# Note that compile times may increase if this is used with big tuples.
+@inline foreach_enumerate(func, collection) = foreach_enumerate(func, collection, 1)
+@inline foreach_enumerate(func, collection::Tuple{}, index) = nothing
+
+@inline function foreach_enumerate(func, collection, index)
+    element = first(collection)
+    remaining_collection = Base.tail(collection)
+
+    func((index, element))
+
+    # Process remaining collection
+    foreach_enumerate(func, remaining_collection, index + 1)
+end
+
 function rhs!(du_ode, u_ode, semi::SemidiscretizationCoupled, t)
     @unpack u_indices = semi
 
     time_start = time_ns()
 
-    @trixi_timeit timer() "copy to coupled boundaries" begin
```

Comment:
Why was the timer removed? Was it added in a different location?

Reply:
This was initially removed due to extra memory allocations. It was one of the suspected places for the type instability. However, I just tested it and it does not lead to extra memory allocations, so it is now back in.

Reply:
OK, great